### ISSUE TITLE
validate a field before ajax request

### ISSUE THREAD

**Issue by danferreira** [limonte/sweetalert2#118]
Hi, I'm switching my project from sweetalert to your lib.

How can a validate a field before an ajax request?

In sweetalert (original) I used closeOnConfirm:false and  showLoaderOnConfirm:true and made my ajax calls on callback.
How can I reproduce this on sa2?
I just want to validate and, if it's correct, make a ajax request with a loader animation.

Edit: in the ajax example of your site, you make your calls in preconfirm I guess, but it is called before inputvalidator.

**Comment by limonte** [comment]
Hi @danferreira! In swal2 there's `preConfirm` parameter which can be set to function which returns Promise. Take a look on AJAX example: http://limonte.github.io/sweetalert2/#ajax-request

**Comment by limonte** [comment]
> Edit: in the ajax example of your site, you make your calls in preconfirm I guess, but it is called before inputvalidator.

Good point, `inputValidator` should be called before `preConfirm`. I'll change this.

**Comment by limonte** [comment]
Done in [v2.2.0](https://github.com/limonte/sweetalert2/releases/tag/v2.2.0)

Take a look on the updated example: http://limonte.github.io/sweetalert2/#ajax-request

**Comment by danferreira** [comment]
Yeah, that's it. 
Thank you. I will update my project and test. I'm happy that your fork exists :D

**Comment by danferreira** [comment]
I dont know if its related but, can I grab the value from input in preconfirm?
In your example, it asks for an email but you not use it in the `preConfirm` function

**Comment by limonte** [comment]
Hi @danferreira ! You can now :) I just published [v2.2.11](https://github.com/limonte/sweetalert2/releases/tag/v2.2.11):

```js
swal({
  ...
  preConfirm: function(value) {
    return new Promise(function(resolve) {
      // use value here
    });
  },
  ...
})
```

**Comment by danferreira** [comment]
worked. Thanks again.
